File: app/src/main/java/de/haukesomm/sokoban/game/Id.java
package de.haukesomm.sokoban.game;

import java.util.Random;

public class Id {

    private static final char[] CHARS = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ".toCharArray();

    public static String next(int length) {
        var idBuilder = new StringBuilder();
        var random = new Random();

        for (int i = 0; i < length; i++) {
            idBuilder.append(CHARS[random.nextInt(length)]);
        }

        return idBuilder.toString();
    }

    public static String next() {
        return next(6);
    }

    private Id() {}
}

File: core/src/commonMain/kotlin/de/haukesomm/sokoban/core/Id.kt
package de.haukesomm.sokoban.core

import kotlin.jvm.JvmOverloads
import kotlin.jvm.JvmStatic

object Id {
    private const val defaultLength = 6
    private val chars = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ".toCharArray()

    @JvmStatic
    @JvmOverloads
    fun next(length: Int = defaultLength) = buildString {
        for (i in 0 until length) {
            append(chars.random())
        }
    }
}